File: graph/representation/list.py
def main():
    # Read number of nodes (n) and edges (m) from input and map converts string to integer
    n, m = map(int, input().split())

    # Initialize adjacency list: a list of empty lists for each node (1-based indexing)
    graph = [[] for _ in range(n + 1)]

    # Read edges and build the adjacency list
    for _ in range(m):
        u, v = map(int, input().split())  # Read an edge between nodes u and v
        graph[u].append(v)  # Add v to the neighbors of u
        graph[v].append(u)  # Add u to the neighbors of v (undirected graph)

    # Print adjacency list for each node
    for i in range(1, n + 1):
        for neighbor in graph[i]:
            print(neighbor, end=" ")  # Print neighbors in one line separated by space
        print()  # New line after each node’s neighbors

if __name__ == "__main__":
    main()

File: graph/representation/list.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, m;
   cin >> n >> m;
    vector<int> graph[n + 1]; // 1-based node indexing
    // There are no elements currently in any list
    // Below is our input edges
    for(int i = 0; i < m; i++) {
        int u, v;
        cin >> u >> v;
        graph[u].push_back(v);
        graph[v].push_back(u);
    }
    for(int i = 1 ; i <= n ; i++){
        for(int j = 0 ; j < graph[i].size(); j++){
            cout << graph[i][j] << " ";
        }
        cout<<"\n";
    }
    return 0;
}